Neos.Http.Factories/Classes/ServerRequestFactoryTrait.php

```diff
@@ -0,0 +1,98 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Http\Factories;
+
+use function GuzzleHttp\Psr7\parse_query;
+use GuzzleHttp\Psr7\ServerRequest;
+use Neos\Flow\Http\Helper\RequestInformationHelper;
+use Psr\Http\Message\ServerRequestFactoryInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UriFactoryInterface;
+
+/**
+ *
+ */
+trait ServerRequestFactoryTrait
+{
+    /**
+     * @var UriFactoryInterface
+     */
+    protected $uriFactory;
+
+    /**
+     * @var string
+     */
+    protected $defaultUserAgent = '';
+
+    /**
+     * @var string
+     */
+    protected $scriptPath = '';
+
+    /**
+     * @var string
+     */
+    protected $defaultHttpVersion = '1.1';
+
+    /**
+     * ServerRequestFactory constructor.
+     *
+     * @param UriFactoryInterface $uriFactory
+     * @param string $defaultUserAgent
+     * @param string $scriptPath
+     * @param string $defaultHttpVersion
+     */
+    public function __construct(
```

Comment:
i just noticed that this trait comes with a constructor which is barely usable in real word ... and an odd piece of code to look at :D Maybe we should fix this somehow?